common/font.go

```diff
@@ -145,3 +146,154 @@ func (f *Font) Render(text string) Texture {
 	imObj := NewImageObject(nrgba)
 	return NewTextureSingle(imObj)
 }
+
+// generateFontAtlas generates the font atlas for this given font, using the first `c` Unicode characters.
+func (f *Font) generateFontAtlas(c int) FontAtlas {
+	atlas := FontAtlas{
+		XLocation: make([]float32, c),
+		YLocation: make([]float32, c),
+		Width:     make([]float32, c),
+		Height:    make([]float32, c),
+	}
+
+	var (
+		int26Width  fixed.Int26_6
+		int26Height fixed.Int26_6
+
+		totalHeight fixed.Int26_6
+
+		fupe        = fixed.Int26_6(f.TTF.FUnitsPerEm())
+		maxYBearing = fixed.Int26_6(0)
+
+		currentX float32
+		maxX     float32
+
+		totalString string // TODO; string is immutable, so this is relatively inefficient; see also where we use `totalString += string(char)`
+		subString   string
+
+		drawCurY int
+	)
+
+	// The "full image"
+	nrgba := image.NewNRGBA(image.Rect(0, 0, 1200, 5000)) // way too big; hopefully
```

Comment:
AFAIK, we can't resize an `image.NRGBA` to whatever size we want to, and we need it to be big enough to fit everything in there. We can't "precompute" the size efficiently, either.